### PR TITLE
feat(multicluster/remote-outline): outline remote service

### PR DESCRIPTION
**Pull request by timmyreilly**
chart and api outline for remoteservice

Signed-off-by: timmyreilly <tireilly@microsoft.com>

**Description**:

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [X] |
| Documentation              | [ ] |
| Install                    | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Ingress                    | [ ] |
| Egress                     | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| SMI Policy                 | [ ] |
| Sidecar Injection          | [ ] |
| Security                   | [ ] |
| Upgrade                    | [ ] |
| Tests                      | [ ] |
| CI System                  | [ ] |
| Demo                       | [ ] |
| Performance                | [ ] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?
No
1. Is this a breaking change?
No

### CODE REVIEW

**Review by shashankram — COMMENTED** [review]
Hey @timmyreilly, thanks for contributing.

To simplify reviews, would you mind breaking this change up as follows:
1. CRD definition
2. Go types and autogenerated code in pkg/apis
3. Helm chart changes (values.yaml and schema)
4. Go code changes in pkg/config
5. Integrate with osm-controller using a feature flag

**Comment by timmyreilly** [comment]
@shashankram can do!

File: charts/osm/values.yaml [hunk 1]
```diff
@@ -169,6 +169,9 @@ OpenServiceMesh:
   # -- Deploy Jaeger during OSM installation
   deployJaeger: false
 
+  # -- Enable multicluster 
+  enableMulticluster: false
```
Comment (snehachhabria) [inline charts/osm/values.yaml:173]: I have already created a feature flag for this here : https://github.com/openservicemesh/osm/blob/4c390a5026874c5e35f52457dc9fb1c818375ea0/charts/osm/values.yaml#L227

Please could we use this

File: charts/osm/crds/remoteservice.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
+{{ if .Values.enableMulticluster }}
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: remoteservices.multicluster.openservicemesh.io
+spec:
+  group: multicluster.openservicemesh.io
```
Comment (snehachhabria) [inline charts/osm/crds/remoteservice.yaml:7]: As discussed offline we want this to be a part of the `config.openservicemesh.io` that already exists

File: charts/osm/crds/remoteservice.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
+{{ if .Values.enableMulticluster }}
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: remoteservices.multicluster.openservicemesh.io
```
Comment (snehachhabria) [inline charts/osm/crds/remoteservice.yaml:5]:
```suggestion
  name: remoteservices.config.openservicemesh.io
```

**Review by snehachhabria — CHANGES_REQUESTED** [review]

File: pkg/announcements/types.go [hunk 4]
```diff
@@ -156,6 +156,15 @@ const (
 
 	// EgressUpdated is the type of announcement emitted when we observe an update to egress.policy.openservicemesh.io
 	EgressUpdated AnnouncementType = "egress-updated"
+
+	// Config
+	ConfigAdded AnnouncementType = "config-added"
+
+	// ConfigDeleted the type of announcement emitted when we observe a deletion of config.policy.openservicemesh.io
+	ConfigDeleted AnnouncementType = "config-deleted"
```
Comment (timmyreilly) [inline pkg/announcements/types.go:164]: To be MultiClusterServiceAdded

File: pkg/apis/config/v1alpha1/register.go [hunk 5]
```diff
@@ -39,6 +39,8 @@ func addKnownTypes(scheme *runtime.Scheme) error {
 	scheme.AddKnownTypes(SchemeGroupVersion,
 		&MeshConfig{},
 		&MeshConfigList{},
+		&RemoteService{},
```
Comment (timmyreilly) [inline pkg/apis/config/v1alpha1/register.go:42]: MulticlusterService + MulticluasterServiceList

File: pkg/kubernetes/types.go [hunk 6]
```diff
@@ -58,6 +58,8 @@ const (
 	Endpoints InformerKey = "Endpoints"
 	// ServiceAccounts lookup identifier
 	ServiceAccounts InformerKey = "ServiceAccounts"
+	// RemoteServices lookup identifier
+	RemoteServices InformerKey = "RemoteServices"
```
Comment (timmyreilly) [inline pkg/kubernetes/types.go:62]: MulticlusterServices InformerKey = "MulticlusterServices

**Pull request closed without merging**